sources/packages/web/src/views/aest/FormioEditor.vue

```diff
@@ -0,0 +1,305 @@
+    const { getReadyToSaveFormDefinition, getFormattedFormDefinition } =
+      useFormioUtils();
+    const saveDynamicFormModal = ref({} as ModalDialog<boolean>);
+    const applyDynamicFormDefinitionModal = ref({} as ModalDialog<boolean>);
+    const formioBuilderRef = ref();
+    const formsListOptionsLoading = ref(false);
+    const formDefinitionLoading = ref(false);
+    const saveDynamicFormLoading = ref(false);
+    const formsListOptions = ref<FormAPIOutDTO[]>([]);
+    const selectedForm = ref<FormAPIOutDTO>();
+    let builder: FormIOBuilder;
+    const formDefinition = ref<string>("");
+    const { generateJSONFileFromContent } = useFileUtils();
+
+    onMounted(async () => {
+      // Load the formio definitions list.
+      formsListOptionsLoading.value = true;
+      try {
+        const list = await ApiClient.DynamicForms.getFormsList();
+        formsListOptions.value = list.forms;
+      } catch {
+        snackBar.error("Unexpected error loading forms list.");
+      } finally {
+        formsListOptionsLoading.value = false;
+      }
+    });
+
+    /**
+     * Load the select form definition.
+     * @param form form to be loaded.
+     */
+    const formSelected = async (form: FormAPIOutDTO): Promise<void> => {
+      if (!form?.name) {
+        return;
+      }
+      formDefinitionLoading.value = true;
+      try {
+        const definition = await ApiClient.DynamicForms.getFormDefinition(
+          form.name,
+        );
+        // Recreate the formio builder.
+        builder = await Formio.builder(formioBuilderRef.value, definition, {
+          noNewEdit: true,
+          noDefaultSubmitButton: true,
+          alwaysConfirmComponentRemoval: true,
+          builder: {
+            custom: FORMIO_CUSTOM_COMPONENTS,
+          },
+        });
+        updateFormDefinitionInspector();
+        builder?.on("change", () => {
+          updateFormDefinitionInspector();
+        });
+      } catch {
+        snackBar.error("Unexpected error loading form definition.");
+      } finally {
+        formDefinitionLoading.value = false;
+      }
+    };
+
+    /**
+     * Update the form definition inspector with the current form definition.
+     */
+    const updateFormDefinitionInspector = async (): Promise<void> => {
+      formDefinition.value = getFormattedFormDefinition(builder.schema);
+    };
+
+    /**
+     * Persists the change to the dynamic form definition database.
+     */
+    const saveDynamicForm = async (): Promise<void> => {
+      if (selectedForm.value?.name === undefined) {
+        snackBar.warn("Please select a form to save.");
+        return;
+      }
+      const shouldContinue = await saveDynamicFormModal.value.showModal();
+      if (!shouldContinue) {
+        return;
+      }
+      try {
+        saveDynamicFormLoading.value = true;
+        await ApiClient.DynamicForms.updateForm(selectedForm.value.name, {
+          formDefinition: getReadyToSaveFormDefinition(builder.schema),
+        });
```

Comment:
It would be better to show a toast when form is saved to the server.

Reply:
Either the condition is present, or we add the null propagation check.
I changed to use the null propagation now.

Reply:
https://github.com/bcgov/SIMS/pull/4674#discussion_r2085248784

Reply:
Toast added.